### PR TITLE
[release-v0.40] Prepare release v0.40.2

### PR DESCRIPTION
**Pull request by rfratto**

### CODE REVIEW

**Review by erikbaranowski — APPROVED** [review]

**Comment by rfratto** [comment]
Holding off on merging due to concerns raised by @jkroepke [on Slack](https://grafana.slack.com/archives/C01050C3D8F/p1709666060262439):

> Hi, I have a follow-up question to https://github.com/grafana/agent/pull/6620, more specific to this line: https://github.com/grafana/agent/pull/6620/files#diff-04a6b7bda5c4ae29d3bcb9ca764f277cbe918089beb04984c9b8826938b00308R179
>
> If INSTDIR is now readable by default to everyone, I may ask myself, if the INSTDIR is a good place for the configuration. Normally, credentials are placed on the config file (e.g.: Grafana Cloud Credentials) and now, they are more readable for everyone. This is expected and should I use environment variables for sensitive values?

**Comment by rfratto** [comment]
I'm planning on pulling in #6622 to address the concerns above after it's merged

**Pull request merged**
